flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterClientFactory.java

```diff
@@ -87,6 +87,6 @@ private YarnClusterDescriptor getClusterDescriptor(Configuration configuration)
                 yarnConfiguration,
                 yarnClient,
                 YarnClientYarnClusterInformationRetriever.create(yarnClient),
-                false);
+                true);
```

Comment:
Why is the `yarnClient` set to shared? That way, it will never be closed.

Reply:
Sorry i forget to revert it.

Reply:
Done. set to false